<launch>

  <arg name="namespace" default="/" />
  <arg name="camera" default="camera" />
  <arg name="respawn" default="False" />
  <arg name="osi_pi" default="False" />
  <arg name="use_proxy" default="False" />
  <arg name="auto_mode" default="False" />
  <arg name="mock" default="False" />

  <node name="pot_state_server_node" pkg="slagsquare_pot_state_server" type="pot_state_server_node.py"
        ns="$(arg namespace)" output="screen" respawn="False">

        <!-- load params -->
    <rosparam file="$(find camera_patrol_controller)/config/$(arg namespace)_$(arg camera)_camera_patrol_controller_params.yaml" />
    <rosparam file="$(find slagsquare_pot_state_server)/config/slagsquare_pot_state_server_params.yaml" />
    <rosparam file="$(find slagsquare_pot_state_server)/config/crane_observer_params.yaml" subst_value="true"/>
    <rosparam file="$(find smelter_osi_pi)/config/umicore_osi_pi_config_params.yaml" />

    <param name="~osi_pi" value="$(arg osi_pi)" />
    <param name="~use_proxy" value="$(arg use_proxy)" />
    <param name="~auto_mode" value="$(arg auto_mode)" />
    <param name="~mock" value="$(arg mock)" />

    <remap from="detected_pots" to="$(arg camera)/rgb/detected_pots" />
    <remap from="ir/image_raw" to="$(arg camera)/ir/snapshot/image_raw" />
    <remap from="ptz/state" to="$(arg camera)/ptz/state" />

    <remap from="ir/detected_pots" to="$(arg camera)/ir/detected_pots" />
    <remap from="ir/detected_pots/image_raw" to="$(arg camera)/ir/detected_pots/image_raw" />

  </node>

</launch>
